src/main/java/com/fleetpin/graphql/dynamodb/manager/InMemoryDynamoDb.java

```diff
@@ -0,0 +1,249 @@
+        this.idGenerator = idGenerator;
+    }
+
+    @Override
+    public <T extends Table> CompletableFuture<T> delete(final String organisationId, final T entity) {
+        return CompletableFuture.supplyAsync(() -> {
+            if (!organisationId.equals(entity.getSourceOrganistaionId())) {
+                return entity;
+            }
+
+            map.remove(new DatabaseKey(organisationId, entity.getClass(), entity.getId()));
+
+            return entity;
+        });
+    }
+
+    @Override
+    public <T extends Table> CompletableFuture<T> deleteLinks(final String organisationId, final T entity) {
+        return CompletableFuture.supplyAsync(() -> {
+            final var databaseKey = new DatabaseKey(organisationId, entity.getClass(), entity.getId());
+
+            final var item = map.get(databaseKey);
+            item.getLinks().clear();
+
+            entity.getLinks().clear();
+
+            return entity;
+        });
+    }
+
+    @Override
+    public <T extends Table> CompletableFuture<T> put(final String organisationId, final T entity) {
+        return CompletableFuture.supplyAsync(() -> {
+            entity.setId(Objects.requireNonNullElseGet(entity.getId(), () -> {
+                entity.setCreatedAt(Instant.now());
+                return newId();
+            }));
+
+            entity.setUpdatedAt(Instant.now());
+
+            final var links = AttributeValue.builder().m(entity.getLinks()
+                    .entries()
+                    .stream()
+                    .collect(Collectors.toMap(
+                            Map.Entry::getKey,
+                            value -> AttributeValue.builder().ss(value.getValue()).build()
+                    ))
+            ).build();
+
+            final var item = new HashMap<String, AttributeValue>();
+            item.put("organisationId", AttributeValue.builder().s(organisationId).build());
+            item.put("id", createTableNamedKey(entity.getClass(), entity.getId()));
+            item.put("item", TableUtil.toAttributes(objectMapper, entity));
+            item.put("links", links);
+            appendSecondaryItemFields(entity, item);
+
+            final var databaseKey = new DatabaseKey(organisationId, entity.getClass(), entity.getId());
+            final var dynamoItem = new DynamoItem(entity.getSourceTable(), item);
+
+            map.put(databaseKey, dynamoItem);
+
+            return entity;
+        });
+    }
+
+    @Override
+    public CompletableFuture<List<DynamoItem>> get(final List<DatabaseKey> keys) {
+        return CompletableFuture.supplyAsync(() -> {
+            final var filtered = getWithFilter(entry -> keys.stream().anyMatch(key -> foundInMap(entry, key)));
+
+            // DataLoader requires the same number of outputs as inputs
+            // Add nice filler
+            for (int i = 0; i < keys.size() - filtered.size(); i++) {
+                filtered.add(null);
+            }
+
+            return filtered;
```

Comment:
offsets need to lineup can't just add on end
query for `[1,2,3]`
and 2 does not exist need to return
`[1response, null, 3response]`